java-util/src/main/java/io/druid/java/util/common/io/smoosh/FileSmoosher.java

```diff
@@ -414,24 +414,28 @@ private Outer getNewCurrOut() throws FileNotFoundException
     final int fileNum = outFiles.size();
     File outFile = makeChunkFile(baseDir, fileNum);
     outFiles.add(outFile);
-    return new Outer(fileNum, new FileOutputStream(outFile), maxChunkSize);
+    return new Outer(fileNum, outFile, maxChunkSize);
   }
 
   public static class Outer implements SmooshedWriter
   {
     private final int fileNum;
     private final int maxLength;
+    private final File outFile;
     private final GatheringByteChannel channel;
 
     private final Closer closer = Closer.create();
     private int currOffset = 0;
 
-    Outer(int fileNum, FileOutputStream output, int maxLength)
+    Outer(int fileNum, File outFile, int maxLength) throws FileNotFoundException
     {
       this.fileNum = fileNum;
-      this.channel = output.getChannel();
+      this.outFile = outFile;
       this.maxLength = maxLength;
-      closer.register(output);
+
+      FileOutputStream outStream = new FileOutputStream(outFile);
+      this.channel = outStream.getChannel();
+      closer.register(outStream);
```

Comment:
can you move the closer to the same place the file output stream is created?

```java
final FileOutputStream outStream = closer.register(new FileOutputStream(outFile));
```

should work